lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1312 @@
+    """
+    Determine sample/specimen collection date from the given REDCap *record*.
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        return record["nasal_swab_q"]
+    elif collection_method == CollectionMethod.SWAB_AND_SEND:
+        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
+    else:
+        return None
+
+def combine_multiple_fields(record: Dict[Any, Any], field_prefix: str, field_suffix: str = "") -> Optional[List]:
+        """
+        Handles the combining of multiple fields asking the same question such
+        as country and state traveled.
+        """
+        regex = rf'^{re.escape(field_prefix)}[0-9]+{re.escape(field_suffix)}$'
+        empty_value = ''
+        answered_fields = list(filter(lambda f: filter_fields(f, record[f], regex, empty_value), record))
+
+        if not answered_fields:
+            return None
+
+        return list(map(lambda x: record[x], answered_fields))
+
+def create_enrollment_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    # Do not include `core_age_years` because we calculate the age ourselves in the computed questionnaire.
+    integer_questions = [
+        'core_zipcode',
+        'weight',
+        'height_ft',
+        'height_in',
+        'height_ft_calc',
+        'height_total',
+        'vaccine_month',
+        'vaccine_year'
+    ]
+
+    string_questions = [
+        'study_area',
+        'text_or_email',
+        'text_or_email_attestation',
+        'campus_location',
+        'student_or_employee',
+        'student_level',
+        'sea_employee_type',
+        'uw_greek_house',
+        'core_home_state',
+        'core_house_members',
+        'dorm',
+        'uw_apt_names',
+        'core_sex',
+        'core_education',
+        'core_income',
+        'vaccine_method',
+        'wfh_base',
+        'core_housing_type',
+        'core_health_risk',
+        'core_tobacco_use',
+        'sought_care_base',
+        'hospital_where_base',
+        'hospital_ed_base',
+        'prior_test_positive_base',
+        'prior_test_type_base',
+        'prior_test_result_base',
+        'contact_base',
+        'wash_hands_base',
+        'clean_surfaces_base',
+        'hide_cough_base',
+        'mask_base',
+        'distance_base',
+        'country1_base',
+        'country2_base',
+        'country3_base',
+        'country4_base',
+        'country5_base',
+        'state1_base',
+        'state2_base',
+        'state3_base',
+        'state4_base',
+        'state5_base',
+        'alerts_off',
+        'bmi',
+        'novax_reason',
+        'covid_vacine',
```

Comment:
Field has been updated to `covid_vaccine`

Reply:
Thanks so much for seeing that. Fixed.

Reply:
I just went through all questionnaires with the latest data dictionary and fixed any discrepancies. I suspect I'll be doing this again before launch.